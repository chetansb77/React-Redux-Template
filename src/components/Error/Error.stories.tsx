import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import Error from ".";

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
export default {
  title: "Component/Error",
  component: Error,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof Error>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
// eslint-disable-next-line react/function-component-definition, react/jsx-props-no-spreading
const Template: ComponentStory<typeof Error> = (args) => <Error {...args} />;

export const ErrorMessage = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
ErrorMessage.args = {};
