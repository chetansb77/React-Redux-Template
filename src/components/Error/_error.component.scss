@import '../../assets/stylesheets/base/colors';

.error-component {
  img {
    @include elementStyle('img');
  }

  .message {
    @include elementStyle('p');
    color: $neutral600;
  }
}