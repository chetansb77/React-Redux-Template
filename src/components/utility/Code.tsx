import React from "react";

interface CodeProps {
  codeSnippet: string;
}

function Code({ codeSnippet }: CodeProps): JSX.Element {
  return (
    <div className="code-snippet">
      {codeSnippet}
    </div>
  );
}

export default Code;
