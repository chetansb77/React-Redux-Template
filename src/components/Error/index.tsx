import React, { useState } from "react";
import Code from "../utility/Code";
import somethingWrongIcon from "../../assets/icons/png/something-wrong.png";

interface ErrorProps {
  errorMessage?: string;
}

function Error({ errorMessage = "" }: ErrorProps): JSX.Element {
  const [showActualError, setShowActualError] = useState(false);

  return (
    <div className="error-component text-center">
      <div>
        <img
          src={somethingWrongIcon}
          className="image-center cursor-pointer"
          alt="something-wrong"
          onClick={() => setShowActualError(!showActualError)}
          aria-hidden="true"
          title="Show error"
        />
      </div>
      <div className="flex justify-center mt-2">
        {showActualError ? (
          <div className="code-message w-auto max-w-md">
            <Code codeSnippet={errorMessage} />
          </div>
        ) : (
          <div className="flex-initial w-72">
            <span className="message">Ooops!! Something went wrong</span>
          </div>
        )}
      </div>
    </div>
  );
}

export default Error;
