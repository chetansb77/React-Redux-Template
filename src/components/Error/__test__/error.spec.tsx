import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Error from "..";
import Code from "../../utility/Code";

Enzyme.configure({ adapter: new Adapter() });

it("error component renders correctly", () => {
  const tree = renderer.create(<Error errorMessage="Error message" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("error component renders correctly when no error message", () => {
  const tree = renderer.create(<Error />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("error component should show code on img click toggle", () => {
  const wrapper = shallow(<Error errorMessage="Error message" />);
  wrapper.find("img").simulate("click");
  expect(wrapper.find(Code)).toHaveLength(1);
});
