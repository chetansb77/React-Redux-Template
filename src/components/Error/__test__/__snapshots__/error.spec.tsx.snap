// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error component renders correctly 1`] = `
<div
  className="error-component text-center"
>
  <div>
    <img
      alt="something-wrong"
      aria-hidden="true"
      className="image-center cursor-pointer"
      onClick={[Function]}
      src=""
      title="Show error"
    />
  </div>
  <div
    className="flex justify-center mt-2"
  >
    <div
      className="flex-initial w-72"
    >
      <span
        className="message"
      >
        Ooops!! Something went wrong
      </span>
    </div>
  </div>
</div>
`;

exports[`error component renders correctly when no error message 1`] = `
<div
  className="error-component text-center"
>
  <div>
    <img
      alt="something-wrong"
      aria-hidden="true"
      className="image-center cursor-pointer"
      onClick={[Function]}
      src=""
      title="Show error"
    />
  </div>
  <div
    className="flex justify-center mt-2"
  >
    <div
      className="flex-initial w-72"
    >
      <span
        className="message"
      >
        Ooops!! Something went wrong
      </span>
    </div>
  </div>
</div>
`;
